new_csaf/csaf/test/base.py

```diff
@@ -1,4 +1,234 @@
-class CsafTest:
-    pass
+from os import stat
+import csaf
+from f16lib.acas import AcasScenario, AcasShieldScenario, AcasScenarioViewer
 
+from collections.abc import Callable
 
+from abc import ABC, abstractmethod
+
+import numpy as np
+import typing
+import GPyOpt
+import GPy
+
+class Optimizer(ABC):
+    """
+    Generic optimizer
+    """
+    @abstractmethod
+    def prepare(self,
+                configuration_space: dict,
+                obj_func: Callable[[[float]], float]):
+        pass
+
+    @abstractmethod
+    def run_optimization(self,
+                         max_iter: int,
+                         max_time: float,
+                         tolerance: float,
+                         verbose: bool = False):
+        pass
+
+class Scenario(ABC):
```

Comment:
CSAF objects like this should a child of `ABC` and [`CsafBase`](https://github.com/GaloisInc/csaf/blob/d704f6a68b06f860d9b60c323249db00641cd9f1/new_csaf/csaf/core/base.py#L9). The idea is to build up a taxonomy that we can leverage for type checking and custom runtimes.